### PR TITLE
Added more examples to show how to use the new React API

### PR DESCRIPTION
**Pull request by zhenlineo**
Examples for integrating the driver's reactive API with Project Reactor and RxJava.

### CODE REVIEW

**Review by michael-simons — COMMENTED** [review]
Looks good to me. I didn't find the docs referencing that, but maybe it's worth writing a short sentence about why / when it's appropriate using `using` vs. `usingWhen` on the publishers.

File: examples/src/test/java/org/neo4j/docs/driver/ExamplesIT.java [hunk 1]
```diff
@@ -488,4 +491,106 @@ void testHostnameVerificationExample()
...
+            // create some 'Product' nodes
+            try ( Session session = neo4j.driver().session() )
+            {
+                session.run(
+                        "UNWIND ['Tesseract', 'Orb', 'Eye of Agamotto'] AS item " +
+                                "CREATE (:Product {id: 0, title: item})" );
+            }
+
+            // read all 'Product' nodes
+            List<String> titles = await( example.readProductTitlesReactor() );
```
Comment (michael-simons) [inline examples/src/test/java/org/neo4j/docs/driver/ExamplesIT.java:510]: I'd use the Project reactors `StepVerifier` for Reactors test and probably similar things if there any for RxJava2.

File: examples/src/main/java/org/neo4j/docs/driver/RxTransactionFunctionExample.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import org.neo4j.driver.summary.ResultSummary;
+
+public class RxTransactionFunctionExample extends BaseApplication
+{
+    public RxTransactionFunctionExample( String uri, String user, String password )
+    {
+        super( uri, user, password );
+    }
+
+    public Publisher<ResultSummary> printAllProductsReactor()
```
Comment (michael-simons) [inline examples/src/main/java/org/neo4j/docs/driver/RxTransactionFunctionExample.java]: Maybe return the actual implementation here? In the end it's what we wanna demonstrate with the example. 
Would also fit better with the tests for the example.

**Comment by zhenlineo** [comment]
There is currently no doc using this examples. Eventually, we shall have a section in docs talking about reactive and that's where these examples will be used. Currently the purpose of this PR is merely to show how the suggested use pattern looks like.

**Pull request merged**